File: SproInstall/SocketPro/samples/performance/csharp/PerformanceStudyWCF/PerfClient/frmPerfClient.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.ServiceModel;
using SocketProAdapter;

namespace PerfClient
{
    public partial class frmPerfClient : Form
    {
        CUPerformanceQuery m_PerfQuery = new CUPerformanceQuery();
        DataTable m_dt;

        public frmPerfClient()
        {
            InitializeComponent();
        }

        private void frmPerfClient_Load(object sender, EventArgs e)
        {
            
        }

        private void btnSQL_Click(object sender, EventArgs e)
        {
            int n;
            MyCallsClient m_myCalls = new MyCallsClient();
            long lStart = m_PerfQuery.Now();
            for (n = 0; n < 100; n++)
            {
                m_dt = m_myCalls.OpenRowset(txtSQL.Text);
            }
            long lDiff = m_PerfQuery.Diff(lStart);
            txtTime.Text = lDiff.ToString();
        }

        private void btnMyEcho_Click(object sender, EventArgs e)
        {
            int n;
            string str;
            MyCallsClient m_myCalls = new MyCallsClient();
            long lStart = m_PerfQuery.Now();
            for (n = 0; n < 10000; n++)
            {
                str = m_myCalls.MyEcho("MyEcho");
            }
            long lDiff = m_PerfQuery.Diff(lStart);
            txtTime.Text = lDiff.ToString();
        }
    }
}

File: SproInstall/SocketPro/samples/performance/vbnet/PerformanceStudyWCF/PerfClient/frmPerfClient.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports System.ServiceModel
Imports SocketProAdapter

Namespace PerfClient
	Public Partial Class frmPerfClient
		Inherits Form
		Private m_PerfQuery As CUPerformanceQuery = New CUPerformanceQuery()
        Private m_dt As DataTable

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub frmPerfClient_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load

        End Sub

        Private Sub btnSQL_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnSQL.Click
            Dim n As Integer
            Dim m_myCalls As MyCallsClient = New MyCallsClient()
            Dim lStart As Long = m_PerfQuery.Now()
            For n = 0 To 99
                m_dt = m_myCalls.OpenRowset(txtSQL.Text)
            Next n
            Dim lDiff As Long = m_PerfQuery.Diff(lStart)
            txtTime.Text = lDiff.ToString()
        End Sub

		Private Sub btnMyEcho_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnMyEcho.Click
			Dim n As Integer
			Dim str As String
			Dim m_myCalls As MyCallsClient = New MyCallsClient()
			Dim lStart As Long = m_PerfQuery.Now()
			For n = 0 To 9999
				str = m_myCalls.MyEcho("MyEcho")
			Next n
			Dim lDiff As Long = m_PerfQuery.Diff(lStart)
			txtTime.Text = lDiff.ToString()
		End Sub
	End Class
End Namespace